### PR TITLE
[MJLINK-6] Add 'sourceJdkModules' configuration parameter.

### PR DESCRIPTION
**Pull request by andretadeu**
This parameter allows to create a distributable Java custom runtime image without relying on JAVA_HOME/Toolschain. This allows to create a Java custom runtime image from one Operating System to another.

I cannot rely on JAVA_HOME and Toolchain location because this JDK have to be able to run where the build will be run. This is not the case when I want to generate a Java custom runtime image for Windows, building the project in a Linux host, for example.

PS: It is not clear where JDK for this issue's test will be located.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MJLINK) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Format the pull request title like `[MJLINK-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MJLINK-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [X] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/it/projects/MJLINK-6_allowSetJmodPath/pom.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          <target>1.9</target>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jlink-plugin</artifactId>
+        <version>@project.version@</version>
+        <extensions>true</extensions>
+        <configuration>
+          <!-- <sourceJdkModules>/home/andre/Desktop/openjdk-11.0.1_windows-x64_bin/jdk-11.0.1</sourceJdkModules> -->
```
Comment (rfscholte) [inline src/it/projects/MJLINK-6_allowSetJmodPath/pom.xml]: We need to think of a better way to verify this. How about

    <sourceJdkModules>${java.home}</sourceJdkModules>

File: src/it/projects/MJLINK-6_allowSetJmodPath/verify.groovy [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+import java.io.*;
+import java.util.*;
+import java.util.jar.*;
+import org.codehaus.plexus.util.*;
+
+boolean result = true;
+
+try
```
Comment (rfscholte) [inline src/it/projects/MJLINK-6_allowSetJmodPath/verify.groovy]: Can be simplified a lot, see comments of https://github.com/apache/maven-dependency-plugin/pull/9

**Comment by andretadeu** [comment]
@rfscholte Done

**Comment by rfscholte** [comment]
Jenkins at ASF is happy too. If you can remove the configuration from maven-compiler-plugin, add yourself as contributor to the pom.xml (if desired) and squash the commits, I'll merge them into master.

**Comment by andretadeu** [comment]
@rfscholte Done

**Pull request merged**